### ISSUE TITLE
Crash in RLMUpdateRealmToSchemaVersion / RLMMigrationRequired but no information in log

### ISSUE THREAD

**Issue by isoiphone** [realm/realm-cocoa#2437]
This is more of a support request than a well defined bug. So apologies in advance!

I'm experiencing a crash on iOS 7.1.1 on iPhone 4 (also have hit it in the simulator). I can't reproduce it on iOS 8.1, but that may be for unrelated reasons.

Both realm 0.94.1 and 0.93.2 experience the same crash. In 0.94.1 the stack trace is not useful at all. It does not go into realm (it stops in the calling file). On 0.93.2 there is significantly more debug information. Has something changed that has affected the amount of debug information available on crash?

There are a few strange, and probably things here:
- On the line of the crash there is an exception generated that has a ton of useful information, but none of that appears in the crash log. Is there a way I can see this?
- This is a fresh install of the app, not on top of an old version. I don't understand why migration is happening at all.
-- this happens when launching app after I logout, which triggers a realm.deleteAllObjects(), maybe this is a factor?
- WhyyyYyyyyy ?

Thanks!

0.93.2 log https://gist.github.com/isoiphone/134aac39627bbb5fe463
0.94.1 log https://gist.github.com/isoiphone/f133bbca5865ee7acbd5

**Comment by isoiphone** [comment]
Crittercism gives me a nice dump of the exception after all. Yay. So disregard that first bullet.

**Comment by isoiphone** [comment]
Resolved this. Changed a handful of things. Spent an hour trying to figure out which of the things was responsible. I'm not 100% sure.

The two most suspect pieces:
- I was caching my realm instance per-thread, to avoid the overhead of creating and configuring it every single time I accessed it.
- I would use a different realm path depending on if encryption was enabled or disabled. It's possible that something else was wrong with this and caused me to access a realm at one path, and then at another. (From a fresh install, so both would be brand new, but perhaps this causes the migration code / dispatch once to fall over?)

@jpsim After ~5 hours on this bug I'm going to just cut my loses, delete all the suspect code, bump to 0.94.1 again and see if it works. If it crashes again I'll post more information here.
